code/game/machinery/recharger.dm

```diff
@@ -163,3 +168,90 @@
 			icon_state = "recharger2"
 		return
 	icon_state = "recharger0"
+
+/obj/machinery/recharger/wallrecharger
+	name = "wall recharger"
+	icon = 'icons/obj/wallcharger.dmi'
+	icon_state = "wrecharger0"
+
+/obj/machinery/recharger/wallrecharger/Initialize()
+	. = ..()
+	var/obj/item/weapon/circuitboard/machine/recharger/wallrecharger/B = new()
+	B.apply_default_parts(src)
+
+/obj/item/weapon/circuitboard/machine/recharger/wallrecharger
+	name = "Weapon Wall Recharger (Machine Board)"
+	build_path = /obj/machinery/recharger/wallrecharger
+	origin_tech = "powerstorage=4;engineering=3;materials=4"
+	req_components = list(/obj/item/weapon/stock_parts/capacitor = 1)

```

Comment:
If you actually want to make it constructable have a second component at least. A matter bin would work.

Also move the construction stuff to the parent

Reply:
Why? It's literally a charger on the wall, it should only take a different board